### ISSUE TITLE
All recorded videos have a 3 seconds freeze in the beginning

### ISSUE THREAD

**Issue by hpicelli** [ccrisan/motioneye#2887]
Hi there,
I'm facing this problem.
In a recorded video (all videos have this issue), it plays one second, freeze for 3 seconds, then plays normal until the end.
I'm using mp4 (h264) now, but tested avi (mpeg4) and the problem persists on all formats i've tested. 
Check attached video.

motionEye Version: 0.43.0
Motion Version: 4.5.1
OS Version: Linux 6.1.0-13-686-pae (Debian 12 32bit)
*Using 32bit OS because i'ts a old machine. This issue is only on 0.43.0, 0.42 runs ok.

Best regards,
Henrique

https://github.com/motioneye-project/motioneye/assets/3333923/72af2417-4b63-44bf-b9bd-77ae739b1e78

**Comment by scott-vincent** [comment]
I have the same problem where first 3 seconds of every recorded video is frozen. Running on Raspberry Pi 5 with 64-bit OS.

**Comment by guaycuru** [comment]
I have the same problem. I thought it was the encoder using too much cpu but changing the code makes no difference

**Comment by hpicelli** [comment]
I think that is a issue on 0.43.
I'll reinstall using 0.42.
This is really annoying.

**Comment by Marijn0** [comment]
Maybe this has something to do with the pre_capture value:
https://motion-project.github.io/4.5.1/motion_config.html#pre_capture

Can someone who is experiencing this issue set the 'Captured Before' in motionEye to 0 and see if it makes a difference?

**Comment by MichaIng** [comment]
Indeed, `pre_capture` is set to 1 (second) by default. Not sure whether it can take 3 seconds to process this 1 second though. At least worth an attempt. Can be changed in motionEye UI.

**Comment by hpicelli** [comment]
The "captured before" and the "captured after" are defined in frames, not
seconds. Any tip? Here "motion gap" is defined to 5 seconds, "Captured
Before" and "Captured After" are defined to 10 frames, and the "Minimum
Motion Frames" are defined to 10 frames.
See attached image.
Best regards.
Henrique
[image: image.png]

**Comment by Marijn0** [comment]
@hpicelli 

What type of camera did you use?
Did you update motioneye to 0.43.0 from 0.42 or a fresh installation?

**Comment by MichaIng** [comment]
> The "captured before" and the "captured after" are defined in frames, not
seconds.

Oh right. Please assign "0" to "captured before", to see if it plays a role at all. "captured after" and "motion gap" do not matter in this regards.

> See attached image.

Images cannot be posted when you answer via email.

**Comment by Marijn0** [comment]
Even if you have set pre_capture=0 there is probably still a buffer:
https://motion-project.github.io/4.5.1/motion_config.html#minimum_motion_frames
But even if this were the problem, it would be more down to motion 🤔

**Comment by MichaIng** [comment]
But that section of the docs at least does not state that one will miss frames until the buffer has been processed. I am not sure how motion handles this internally, so worth a try to disable pre-capture, where this is mentioned explicitly. But if it does not help, also setting minimum motion frames to "1" could be tested. motionEye sets this to 20 by default, but can be adjusted in web UI as well.

**Comment by hpicelli** [comment]
> @hpicelli
> 
> What type of camera did you use? Did you upgrade motioneye to 0.43.0 from 0.42 or a fresh installation?

Hi.
Fresh 0.43.0. Upgraded to 0.43.1b2 yesterday. Same issue.
Cameras are generic. In 0.42 does't have this problem.

**Comment by Marijn0** [comment]
@hpicelli
Have you tried assigning 0 to captured before??
And 1 to minimum motion frames??
And if this doesn't solve the problem, please also try turning on Movie Passthrough.

**Comment by hpicelli** [comment]
Hi folks.
capture before set to 0: the problem stills (@MichaIng informed that maybe the 0 = 20 (default by motion))
capture before set to 1: the problem stills
Both options tested after system reboot.

@Marijn0

**Comment by hpicelli** [comment]
@Marijn0

Activating the button still causes the same problem.
No changes so far have changed the problem at all.
To ensure this, the system is restarted with every change made.

**Comment by MichaIng** [comment]
Checking some open issues: @hpicelli did you try to set "minimum motion frames" to 1? You said "capture before set to 1: the problem stills", probably a typo?

**Comment by TedBusto** [comment]
I'm also experiencing this issue. .mkv file format and my Capture Before set way too high but still the same 3 second freeze.

Raspberry Pi 3B+
motionEye Version 	0.43.1
Motion Version 	4.7.1
OS Version 	Debian 12

**Comment by Marijn0** [comment]
@TedBusto 
Have you tried setting Capture Before to 0, and does the freeze still occur?
